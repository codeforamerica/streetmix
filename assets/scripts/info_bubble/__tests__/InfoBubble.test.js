/* eslint-env jest */
import React from 'react'
import { InfoBubble } from '../InfoBubble'
import { shallow } from 'enzyme'

jest.mock('../../streets/data_model', () => {})
jest.mock('../../segments/view')

describe('InfoBubble', () => {
  it('renders', () => {
    const segment = {}
    const street = {
      segments: [segment]
    }
    const locale = { locale: 'en' }
    const wrapper = shallow(<InfoBubble locale={locale} street={street} position={0} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('shows description', () => {
    const street = {
      segments: []
    }
    const locale = { locale: 'en' }
    const wrapper = shallow(<InfoBubble locale={locale} street={street} position={'u'} descriptionVisible />)
    expect(wrapper).toMatchSnapshot()
  })
  it('is visible', () => {
    const street = {
      segments: []
    }
    const locale = { locale: 'en' }
    const wrapper = shallow(<InfoBubble locale={locale} street={street} position={'u'} visible />)
    expect(wrapper).toMatchSnapshot()
  })
  it('shows building left info bubble', () => {
    const street = {
      segments: [],
      leftBuildingVariant: 'grass'
    }
    const locale = { locale: 'en' }
    const wrapper = shallow(<InfoBubble locale={locale} street={street} position={'left'} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('shows building right info bubble', () => {
    const street = {
      segments: [],
      rightBuildingVariant: 'grass'
    }
    const locale = { locale: 'en' }
    const wrapper = shallow(<InfoBubble locale={locale} street={street} position={'right'} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('updates hover polygon', () => {
    const segment = {}
    const street = {
      segments: [segment]
    }
    const updateHoverPolygon = jest.fn()
    const locale = { locale: 'en' }
    const wrapper = shallow(<InfoBubble locale={locale} street={street} position={0} visible={false} updateHoverPolygon={updateHoverPolygon} />)
    wrapper.setProps({ visible: true })
    expect(updateHoverPolygon).toHaveBeenCalledTimes(1)
  })
  describe('interactions', () => {
    it('set info bubble mouse inside', () => {
      const street = {
        segments: []
      }
      const locale = { locale: 'en' }
      const setInfoBubbleMouseInside = jest.fn()
      const updateHoverPolygon = jest.fn()
      const wrapper = shallow(<InfoBubble locale={locale} street={street} position={0} visible={false} setInfoBubbleMouseInside={setInfoBubbleMouseInside} updateHoverPolygon={updateHoverPolygon} />)
      wrapper.simulate('mouseenter')
      expect(setInfoBubbleMouseInside).toHaveBeenCalledTimes(1)
      expect(setInfoBubbleMouseInside).toHaveBeenCalledWith(true)
    })
    it('does not set info bubble mouse inside', () => {
      const street = {
        segments: []
      }
      const locale = { locale: 'en' }
      const setInfoBubbleMouseInside = jest.fn()
      const updateHoverPolygon = jest.fn()
      const wrapper = shallow(<InfoBubble locale={locale} street={street} position={0} visible={false} setInfoBubbleMouseInside={setInfoBubbleMouseInside} updateHoverPolygon={updateHoverPolygon} />)
      wrapper.simulate('mouseleave')
      expect(setInfoBubbleMouseInside).toHaveBeenCalledTimes(1)
      expect(setInfoBubbleMouseInside).toHaveBeenCalledWith(false)
    })
  })
})
