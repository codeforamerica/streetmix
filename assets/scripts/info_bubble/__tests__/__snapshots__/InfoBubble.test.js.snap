// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBubble is visible 1`] = `
<div
  className="info-bubble info-bubble-type-segment visible"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
>
  <Triangle
    highlight={false}
  />
  <header>
    <Connect(EditableLabel)
      label={null}
      position={2}
      segment={Object {}}
    />
    <InjectIntl(RemoveButton)
      segment={2}
    />
  </header>
  <div
    className="info-bubble-controls"
  >
    <IntlProvider
      locale="en"
    >
      <InjectIntl(Connect(Variants))
        position={2}
        type={1}
      />
    </IntlProvider>
    <InjectIntl(Connect(WidthControl))
      position={2}
    />
  </div>
  <Warnings
    segment={Object {}}
  />
  <Connect(Description)
    highlightTriangle={[Function]}
    infoBubbleEl={null}
    unhighlightTriangle={[Function]}
    updateBubbleDimensions={[Function]}
    updateHoverPolygon={[Function]}
  />
</div>
`;

exports[`InfoBubble renders 1`] = `
<div
  className="info-bubble info-bubble-type-segment"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
>
  <Triangle
    highlight={false}
  />
  <header>
    <Connect(EditableLabel)
      label={
        <IntlProvider
          locale="en"
        >
          <FormattedMessage
            defaultMessage="Unknown"
            id="segments.undefined"
            values={Object {}}
          />
        </IntlProvider>
      }
      position={0}
      segment={Object {}}
    />
    <InjectIntl(RemoveButton)
      segment={0}
    />
  </header>
  <div
    className="info-bubble-controls"
  >
    <IntlProvider
      locale="en"
    >
      <InjectIntl(Connect(Variants))
        position={0}
        type={1}
      />
    </IntlProvider>
    <InjectIntl(Connect(WidthControl))
      position={0}
    />
  </div>
  <Warnings
    segment={Object {}}
  />
  <Connect(Description)
    highlightTriangle={[Function]}
    infoBubbleEl={null}
    unhighlightTriangle={[Function]}
    updateBubbleDimensions={[Function]}
    updateHoverPolygon={[Function]}
  />
</div>
`;

exports[`InfoBubble shows building left info bubble 1`] = `
<div
  className="info-bubble info-bubble-type-building"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
>
  <Triangle
    highlight={false}
  />
  <header>
    <Connect(EditableLabel)
      label={
        <IntlProvider
          locale="en"
        >
          <FormattedMessage
            defaultMessage="Grass"
            id="buildings.grass.name"
            values={Object {}}
          />
        </IntlProvider>
      }
      position="left"
      segment={Object {}}
    />
  </header>
  <div
    className="info-bubble-controls"
  >
    <IntlProvider
      locale="en"
    >
      <InjectIntl(Connect(Variants))
        position="left"
        type={2}
      />
    </IntlProvider>
    <InjectIntl(Connect(BuildingHeightControl))
      position="left"
    />
  </div>
  <Warnings
    segment={Object {}}
  />
  <Connect(Description)
    highlightTriangle={[Function]}
    infoBubbleEl={null}
    unhighlightTriangle={[Function]}
    updateBubbleDimensions={[Function]}
    updateHoverPolygon={[Function]}
  />
</div>
`;

exports[`InfoBubble shows building right info bubble 1`] = `
<div
  className="info-bubble info-bubble-type-building"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
>
  <Triangle
    highlight={false}
  />
  <header>
    <Connect(EditableLabel)
      label={
        <IntlProvider
          locale="en"
        >
          <FormattedMessage
            defaultMessage="Grass"
            id="buildings.grass.name"
            values={Object {}}
          />
        </IntlProvider>
      }
      position="right"
      segment={Object {}}
    />
  </header>
  <div
    className="info-bubble-controls"
  >
    <IntlProvider
      locale="en"
    >
      <InjectIntl(Connect(Variants))
        position="right"
        type={3}
      />
    </IntlProvider>
    <InjectIntl(Connect(BuildingHeightControl))
      position="right"
    />
  </div>
  <Warnings
    segment={Object {}}
  />
  <Connect(Description)
    highlightTriangle={[Function]}
    infoBubbleEl={null}
    unhighlightTriangle={[Function]}
    updateBubbleDimensions={[Function]}
    updateHoverPolygon={[Function]}
  />
</div>
`;

exports[`InfoBubble shows description 1`] = `
<div
  className="info-bubble info-bubble-type-segment show-description"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
>
  <Triangle
    highlight={false}
  />
  <header>
    <Connect(EditableLabel)
      label={null}
      position={1}
      segment={Object {}}
    />
    <InjectIntl(RemoveButton)
      segment={1}
    />
  </header>
  <div
    className="info-bubble-controls"
  >
    <IntlProvider
      locale="en"
    >
      <InjectIntl(Connect(Variants))
        position={1}
        type={1}
      />
    </IntlProvider>
    <InjectIntl(Connect(WidthControl))
      position={1}
    />
  </div>
  <Warnings
    segment={Object {}}
  />
  <Connect(Description)
    highlightTriangle={[Function]}
    infoBubbleEl={null}
    unhighlightTriangle={[Function]}
    updateBubbleDimensions={[Function]}
    updateHoverPolygon={[Function]}
  />
</div>
`;
